### ISSUE TITLE
[Vanilla Bug] Some terrain transformations (or at least the "hill"-tool) produce faulty tiles

### ISSUE THREAD

**Issue by TheRedDaemon** [gynt/stronghold-mapeditortools#13]
I noticed this first while I had the "shape" feature activated, but this also applies to the vanilla game:

At least if the "hill"-tool is used in a line, then the produced tiles are partially wrong. Most notable the backside of a diagonal hill. This is only fixed if the player rotates the view in the editor. If this is not solved this way, then the tile issues persist into the actual game.
Outside of the editor, rotating does not fix this anymore.

Considering that this feels like a rather involved issue, it might be out of scope, but I wanted to leave it as a note regardless.

**Comment by gynt** [comment]
I knew something like this would come up at some point...
It is the main reason I am replaying button commands: I cannot be blamed for this! You should get the same results if you click your shape manually in the editor.

There is a trick to redraw tiles (telling the game to rotate to the current position, tricking it into a redraw), it might solve it.

**Comment by TheRedDaemon** [comment]
> I cannot be blamed for this! You should get the same results if you click your shape manually in the editor.

I do. I actually made sure it is an issue of the unmodified game to not accidentally blame you. 😄
But I also thought this could be a good place for an issue regarding the map editor.
Do you think I should copy this issue over to the UCP repo?

> There is a trick to redraw tiles (telling the game to rotate to the current position, tricking it into a redraw), it might solve it.

Nice.
However what should trigger it might be a problem (maybe?). It would need to be called after the game has finished applying the current requested modifications.
Another one/two console commands to activate/deactivate might also be needed. Maybe with default to active? Depends on the performace impact, I guess.

**Comment by gynt** [comment]
Yes it requires a larger framework than I currently implemented (in this sense this issue is related to the simultaneous command limit).
